### DISCUSSION TITLE
Xiangqi-Stockfish fork?

### DISCUSSION THREAD

**Discussion by maksimKorzh** [fairy-stockfish/Fairy-Stockfish#277] [Ideas]
Hey what's up guys, Code Monkey King's here.
I did a couple of xiangqi related projects:
Xiangqi engine in JS: https://github.com/maksimKorzh/wukong-xiangqi
Cross-platform XIangqi GUI (electron js): https://github.com/maksimKorzh/ccbridge-js
And has been covering them on youtube: https://www.youtube.com/channel/UCB9-prLkPwgvlKKqDgXhsMQ
But all that is just a proof-of-concept implementations with the didactic purposes in mind,
so now I'm looking for a solid project to contribute and like-minded people to work with on it.

**Comment by mtaktikos** [comment]
The only Xiangqi engine except Fairy-SF, that is declared by its author as SF-based, is Chameleon (https://github.com/EterCyber/Chameleon). It is based on SF7 at so specialized to the Xiangqi game, that it plays still a bit stronger than Fairy-SF. Although Chameleon is rather old, it is not impossible that ideas in its code can be adapted to make Fairy-SF stronger, without to break the other variants

**Reply by maksimKorzh** [reply]
@mtaktikos yeah I know and already tinkering with that for a couple of weeks. I'm now trying to port my JS move generator for xiangqi to C++ and connect it to SF. Surprisingly my JS movegen is even slightly faster than Chameleon's, here's a video showcasing that: https://www.youtube.com/watch?v=tA5--WtJL0I

**Reply by mtaktikos** [reply]
That is promising, keep up the good work!

**Reply by maksimKorzh** [reply]
@mtaktikos thanks for your feedback. I'm suffering from C++ specific stuff (I really used to C) but hopefully when I get used to the environment the work should go faster. I'm pretty clear conceptually how the modules interact with each other within the engine. In sf it's done way more complicated than I used to but the code is very clean and readable. So my current strategy is try not to brake anything while adding new feature, for instance I've just added board array for xiangqi into Position class and altering methods to set fen/print pos object. But anyway I think https://github.com/maksimKorzh/xiangqi-stockfish/issues is a better place for these specific discussions. I'll be covering code updates within the videos and any feedbacks/hints are highly appreciated especially from C++ programmers experienced in chess programming. Cheers!
